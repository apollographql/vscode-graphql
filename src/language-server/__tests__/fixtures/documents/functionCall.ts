declare function gql<A = any, B = any>(arg: string): void;
declare type SomeResult = any;
declare type SomeVariables = any;

// prettier-ignore
gql(`query { test }`)

// prettier-ignore
gql<SomeResult, SomeVariables>(`query { test }`)

// prettier-ignore
gql<{
  test: string;
},{
  test: string;
}>(`query { test }`)

// prettier-ignore
gql(
`query { test }`
)

// prettier-ignore
gql
(
  `query { test }`
  )

// prettier-ignore
gql  (  `query { test }` )

gql(`
  query {
    test
  }
`);

gql<SomeResult, SomeVariables>(`
  query {
    test
  }
`);

gql<
  {
    test: string;
  },
  {
    test: string;
  }
>(`
  query {
    test
  }
`);

// prettier-ignore
gql(
`
  query {
    test
  }
`
);

// prettier-ignore
gql (  `
  query {
    test
  }
` );

export {};
