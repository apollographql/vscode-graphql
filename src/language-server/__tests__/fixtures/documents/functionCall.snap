>declare function gql<A = any, B = any>(arg: string): void;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>declare type SomeResult = any;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>declare type SomeVariables = any;
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql(`query { test }`)
#^^^ inline.graphql apollo.functionCall entity.name.function.tagged-template.js
#   ^ inline.graphql apollo.functionCall
#    ^ inline.graphql apollo.functionCall punctuation.definition.string.template.begin.js
#     ^^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql keyword.operation.graphql
#          ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#           ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#            ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#             ^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
#                 ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#                  ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                   ^^ inline.graphql apollo.functionCall punctuation.definition.string.template.end.js
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql<SomeResult, SomeVariables>(`query { test }`)
#^^^ inline.graphql apollo.functionCall entity.name.function.tagged-template.js
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql apollo.functionCall
#                               ^ inline.graphql apollo.functionCall punctuation.definition.string.template.begin.js
#                                ^^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql keyword.operation.graphql
#                                     ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#                                      ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                                       ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#                                        ^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
#                                            ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#                                             ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
#                                              ^^ inline.graphql apollo.functionCall punctuation.definition.string.template.end.js
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql<{
#^^^^^^ inline.graphql
>  test: string;
#^^^^^^^^^^^^^^^^ inline.graphql
>},{
#^^^^ inline.graphql
>  test: string;
#^^^^^^^^^^^^^^^^ inline.graphql
>}>(`query { test }`)
#^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql(
#^^^^^ inline.graphql
>`query { test }`
#^^^^^^^^^^^^^^^^^ inline.graphql
>)
#^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql
#^^^^ inline.graphql
>(
#^^ inline.graphql
>  `query { test }`
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>  )
#^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql  (  `query { test }` )
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql
>
>gql(`
#^^^ inline.graphql apollo.functionCall entity.name.function.tagged-template.js
#   ^ inline.graphql apollo.functionCall
#    ^ inline.graphql apollo.functionCall punctuation.definition.string.template.begin.js
>  query {
#^^ inline.graphql apollo.functionCall meta.embedded.block.graphql
#  ^^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql keyword.operation.graphql
#       ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#        ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>    test
#^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#    ^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
>  }
#^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#  ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>`);
#^^ inline.graphql apollo.functionCall punctuation.definition.string.template.end.js
#  ^^ inline.graphql
>
>gql<SomeResult, SomeVariables>(`
#^^^ inline.graphql apollo.functionCall entity.name.function.tagged-template.js
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline.graphql apollo.functionCall
#                               ^ inline.graphql apollo.functionCall punctuation.definition.string.template.begin.js
>  query {
#^^ inline.graphql apollo.functionCall meta.embedded.block.graphql
#  ^^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql keyword.operation.graphql
#       ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#        ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>    test
#^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#    ^^^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
>  }
#^^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql
#  ^ inline.graphql apollo.functionCall meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
>`);
#^^ inline.graphql apollo.functionCall punctuation.definition.string.template.end.js
#  ^^ inline.graphql
>
>gql<
#^^^^^ inline.graphql
>  {
#^^^^ inline.graphql
>    test: string;
#^^^^^^^^^^^^^^^^^^ inline.graphql
>  },
#^^^^^ inline.graphql
>  {
#^^^^ inline.graphql
>    test: string;
#^^^^^^^^^^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>>(`
#^^^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>`);
#^^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql(
#^^^^^ inline.graphql
>`
#^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>`
#^^ inline.graphql
>);
#^^^ inline.graphql
>
>// prettier-ignore
#^^^^^^^^^^^^^^^^^^^ inline.graphql
>gql (  `
#^^^^^^^^^ inline.graphql
>  query {
#^^^^^^^^^^ inline.graphql
>    test
#^^^^^^^^^ inline.graphql
>  }
#^^^^ inline.graphql
>` );
#^^^^^ inline.graphql
>